src/unix/signal.c

```diff
@@ -63,6 +74,22 @@ static void uv__signal_global_init(void) {
 }
 
 
+static void uv__signal_global_reinit(void) {
+  /* We can only use signal-safe functions here.
+   * That includes read/write and close, fortunately.
+   * We do all of this directly here instead of resetting
+   * uv__signal_global_init_guard because
+   * uv__signal_global_once_init is only called from uv_loop_init
+   * and this needs to function in existing loops.
+   */
+  uv__close(uv__signal_lock_pipefd[0]);
+  uv__signal_lock_pipefd[0] = -1;
+  uv__close(uv__signal_lock_pipefd[1]);
+  uv__signal_lock_pipefd[1] = -1;
+  uv__signal_global_init();
+}
```

Comment:
Okay, let's leave it the way it is.

> if we're concerned that the uv__signal_global_init_guard pthread_once_t value will somehow get reset back to PTHREAD_ONCE_INIT in the child process

Not exactly PTHREAD_ONCE_INIT but LinuxThreads and the old OpenBSD pthreads implementation too IIRC stored the PID, effectively making pthread_once_t variables reset after a fork.  I don't know if any modern pthreads libraries do that, but the enhancement I suggested would catch that.